### ISSUE TITLE
Printing epoch progress

### ISSUE THREAD

**Issue by rasbt** [Lightning-AI/litgpt#1015]
Now that we specify epochs via the new dataset API, maybe it would be nice to print the current epoch during training. 

Right now it's something like 

```

### Instruction:
Write a response that categorizes your own movies so
iter 100 | step 100: loss 2.1253, iter time: 329.54 ms (optimizer.step)
iter 200 | step 200: loss 1.4829, iter time: 500.94 ms (optimizer.step)
iter 300 | step 300: loss 1.2110, iter time: 505.04 ms (optimizer.step)
iter 400 | step 400: loss 1.1430, iter time: 472.31 ms (optimizer.step)
iter 500 | step 500: loss 1.2097, iter time: 561.45 ms (optimizer.step)
iter 600 | step 600: loss 0.8361, iter time: 389.05 ms (optimizer.step)
Validating ...
Recommend a movie for me to watch during the weekend and explain the reason.
Below is an instruction that describes a task. Write a response that appropriately completes the request.

### Instruction:
Recommend a movie for me to watch during the weekend and explain the reason.

### Response:
In order to recommend a movie to you, I will need some background information about your preferences. Before diving into the recommendation, here are some questions to get you thinking:

1. What are some of my favorite movies?
2. What genres do I usually watch?
3. What is my average viewing time for each movie?
4. What are my top five favorite movies?

Now, let's continue to dig deeper into the recommendation
iter 600: val loss 1.1043, val time: 11987.78 ms
iter 700 | step 700: loss 1.1207, iter time: 616.54 ms (optimizer.step)
iter 800 | step 800: loss 1.0382, iter time: 577.98 ms (optimizer.step)
iter 900 | step 900: loss 1.2909, iter time: 414.35 ms (optimizer.step)
```

It could be 

```

### Instruction:
Write a response that categorizes your own movies so
Ep 01 | iter 100 | step 100: loss 2.1253, iter time: 329.54 ms (optimizer.step)
Ep 01 | iter 200 | step 200: loss 1.4829, iter time: 500.94 ms (optimizer.step)
Ep 01 | iter 300 | step 300: loss 1.2110, iter time: 505.04 ms (optimizer.step)
Ep 01 | iter 400 | step 400: loss 1.1430, iter time: 472.31 ms (optimizer.step)
Ep 01 | iter 500 | step 500: loss 1.2097, iter time: 561.45 ms (optimizer.step)
Ep 01 | iter 600 | step 600: loss 0.8361, iter time: 389.05 ms (optimizer.step)
Validating ...
Recommend a movie for me to watch during the weekend and explain the reason.
Below is an instruction that describes a task. Write a response that appropriately completes the request.

### Instruction:
Recommend a movie for me to watch during the weekend and explain the reason.

### Response:
In order to recommend a movie to you, I will need some background information about your preferences. Before diving into the recommendation, here are some questions to get you thinking:

1. What are some of my favorite movies?
2. What genres do I usually watch?
3. What is my average viewing time for each movie?
4. What are my top five favorite movies?

Now, let's continue to dig deeper into the recommendation
Ep 01 | iter 600: val loss 1.1043, val time: 11987.78 ms
Ep 01 | iter 700 | step 700: loss 1.1207, iter time: 616.54 ms (optimizer.step)
Ep 01 | iter 800 | step 800: loss 1.0382, iter time: 577.98 ms (optimizer.step)
Ep 01 | iter 900 | step 900: loss 1.2909, iter time: 414.35 ms (optimizer.step)
```

Regarding the line length, maybe we can get rid of `(optimizer.step)`? What do you think @awaelchli @carmocca ?

**Comment by carmocca** [comment]
I'm neutral about adding epoch. But I would keep `(optimizer.step)` if gradient accumulation is enabled. I agree that it's not very informative when every print has it.
